pkg/multi-tenancy/config/config.go

```diff
@@ -0,0 +1,37 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package config
+
+// TenantLabelKey is a label key reserved for multi-tenancy.
+const TenantLabelKey = "__tenant__"
+
+// Config defines the configuration for multi-tenancy.
+type Config struct {
+	ValidTenants []string
+	tenantsCache map[string]struct{}
+}
+
+// FillTenantCache creates a cache of valid tenants for multi-tenancy.
+func (cfg *Config) FillTenantCache() {
```

Comment:
instead of this function I suggest a NewConfig(tenantList []string) function that always create a valid Config object. That way you can't have an invalid object without the map filled in being made.

Reply:
Yes, I was thinking of the same as part of optimization. Plus, the caller does not have to remember to call FillTenantCache() as it gets handled automatically.